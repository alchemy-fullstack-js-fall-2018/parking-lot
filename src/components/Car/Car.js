import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import ROUTES from '../../routes';


const Car = ({ car }) => {
  const { plate, make, model, color } = car;
  return (
    <div>
      <p><Link to={ROUTES.CAR.linkTo(plate)}>{plate}</Link>: {color} {make} {model} </p>
    </div>
  );
};

Car.propTypes = {
  car: PropTypes.object.isRequired
};

export default Car;
