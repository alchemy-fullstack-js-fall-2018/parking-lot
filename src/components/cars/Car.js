import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';

const Car = ({ cars }) => {

  const carItems = cars.map(car => {
    return (
      <tr key={car.plate}>
        <td id="make">{car.make}</td>
        <td id="model">{car.model}</td>
        <td id="plate"><Link to={ROUTES.CAR_DETAIL.linkTo(car.plate)}>{car.plate}</Link></td>
      </tr>
    );
  });
  

  return (
    <Fragment>
      {carItems}
    </Fragment>
  );
};

Car.propTypes = {
  cars: PropTypes.array.isRequired
};

export default Car;
