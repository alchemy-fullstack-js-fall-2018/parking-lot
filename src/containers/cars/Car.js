import { connect } from 'react-redux';
import Car from '../../components/Car/Car';
import { getCar } from '../../selectors/cars';

const mapStateToProps = (state, props) => {
  return {
    car: getCar(state, props.match.params.id)
  };
};

export default connect(
  mapStateToProps
)(Car);
