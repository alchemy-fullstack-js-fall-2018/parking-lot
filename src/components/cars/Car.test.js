import React from 'react';
import { shallow } from 'enzyme';
import Car from './Car';
jest.mock('../../routes/routes', () => {
  return {
    ROUTES: {
      CAR_DETAIL: {
        linkTo: () => 'asldkfj'
      }
    }
  };
});

describe('Car component', () => {

  const cars = [{ make: 'Toyota', model: 'Corolla', plate: 'ABC123' }];

  it('matches snapshot', () => {
    const wrapper = shallow(<Car cars={cars}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('displays make, model, and plate of a car', () => {
    const wrapper = shallow(<Car cars={cars} />);

    expect(wrapper.find('#make').props().children).toMatch(/Toyota/);
    expect(wrapper.find('#model').props().children).toMatch(/Corolla/);
    expect(wrapper.find('#plate Link').props().children).toMatch(/ABC123/);
  });
});
