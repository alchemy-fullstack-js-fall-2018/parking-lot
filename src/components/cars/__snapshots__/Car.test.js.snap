// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Car component matches snapshot 1`] = `
<Fragment>
  <tr
    key="ABC123"
  >
    <td
      id="make"
    >
      Toyota
    </td>
    <td
      id="model"
    >
      Corolla
    </td>
    <td
      id="plate"
    >
      <Link
        replace={false}
        to="asldkfj"
      >
        ABC123
      </Link>
    </td>
  </tr>
</Fragment>
`;
